File: spring-web-java/src/main/java/spring/web/java/infrastructure/persistence/adapter/RequestLockRepositoryAdapter.java
package spring.web.java.infrastructure.persistence.adapter;

import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import spring.web.java.domain.repository.RequestLockRepository;
import spring.web.java.infrastructure.persistence.RequestLockRedisRepository;

@Component
@RequiredArgsConstructor
public class RequestLockRepositoryAdapter implements RequestLockRepository {

	private final RequestLockRedisRepository redisRepository;

	@Override
	public boolean setIfAbsent(Long memberId, String method, String uri) {
		return redisRepository.setIfAbsent(memberId, method, uri);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/adapter/RequestLockRepositoryAdapter.kt
package spring.web.kotlin.infrastructure.persistence.adapter

import org.springframework.stereotype.Component
import spring.web.kotlin.domain.repository.RequestLockRepository
import spring.web.kotlin.infrastructure.persistence.RequestLockRedisRepository

@Component
class RequestLockRepositoryAdapter(
    private val redisRepository: RequestLockRedisRepository
) : RequestLockRepository {
    override fun setIfAbsent(memberId: Long, method: String, uri: String): Boolean {
        return redisRepository.setIfAbsent(memberId, method, uri)
    }
}